src/main/java/org/icgc/argo/workflow_management/service/NextflowService.java

```diff
@@ -0,0 +1,185 @@
+    cmdParams.put("params", params.getWorkflowParams());
+
+    // K8s options from application.yml
+    cmdParams.put("namespace", k8sConfig.getNamespace());
+    cmdParams.put("volMounts", Collections.singletonList(k8sConfig.getVolMounts()));
+
+    // Where to POST event-based logging
+    cmdParams.put("withWebLog", webLogUrl);
+
+    // Dynamic engine properties/config
+    val workflowEngineOptions = params.getWorkflowEngineParameters();
+
+    // Use revision if provided in workflow_engine_options
+    if (nonNull(workflowEngineOptions)) {
+
+      if (nonNull(workflowEngineOptions.getWorkflowVersion())) {
+        cmdParams.put("revision", workflowEngineOptions.getWorkflowVersion());
+      }
+
+      // Process options (default docker container to run for process if not specified)
+      if (nonNull(workflowEngineOptions.getDefaultContainer())) {
+        val processOptions = new HashMap<String, String>();
+        processOptions.put("container", workflowEngineOptions.getDefaultContainer());
+        cmdParams.put("process", processOptions);
+      }
+    }
+
+    return createWithReflection(CmdKubeRun.class, cmdParams)
+        .orElseThrow(NextflowReflectionException::new);
+  }
+
+  private K8sDriverLauncher createDriver(@NonNull CmdKubeRun cmd)
+      throws NextflowReflectionException {
+
+    Method checkRunName = null;
+
+    try {
+      checkRunName = CmdKubeRun.class.getDeclaredMethod("checkRunName");
+    } catch (NoSuchMethodException e) {
+      e.printStackTrace();
+    }
+
+    if (checkRunName != null) {
+      checkRunName.setAccessible(true);
+
+      try {
+        checkRunName.invoke(cmd);
+      } catch (IllegalAccessException | InvocationTargetException e) {
+        e.printStackTrace();
+      }
+    } else {
+      throw new NextflowReflectionException("Cannot access checkRunName!");
+    }
+
+    val k8sDriverLauncherParams = new HashMap<String, Object>();
+    k8sDriverLauncherParams.put("cmd", cmd);
+    k8sDriverLauncherParams.put("runName", cmd.getRunName());
+    k8sDriverLauncherParams.put("background", true);
+
+    return createWithReflection(K8sDriverLauncher.class, k8sDriverLauncherParams)
+        .orElseThrow(NextflowReflectionException::new);
+  }
+
+  private <T> Optional<T> createWithReflection(Class<T> objClass, Map<String, Object> params) {
+    try {
+      T obj = objClass.newInstance();
+      return Optional.of(reflectionFactory(objClass, obj, params));
+    } catch (InstantiationException | IllegalAccessException e) {
+      log.error(e.getMessage());
+    }
+
+    return Optional.empty();
+  }
+
+  private <T> T reflectionFactory(Class<T> objClass, T obj, Map<String, Object> map) {
+    map.forEach(
+        (key, value) -> {
+          val field = ReflectionUtils.findField(objClass, key);
+
+          if (nonNull(field)) {
+            ReflectionUtils.makeAccessible(field);
+            ReflectionUtils.setField(field, obj, value);
+          }
+        });
+
+    return obj;
+  }
```

Comment:
2 things:
1. since these have no state, they can be made static
2. even though they are used once, they are still cluttering up the code. I suggest creating a `util` package and creating `Reflections.java` file in there, and moving these static (after doing 1) in there.